src/main/java/net/atomique/ksar/AllParser.java

```diff
@@ -25,13 +25,16 @@
     {
       put("^\\d{8}$", "yyyyMMdd");
       put("^\\d{1,2}-\\d{1,2}-\\d{4}$", "dd-MM-yyyy");
-      put("^\\d{4}-\\d{1,2}-\\d{1,2}$", "yyyy-MM-dd");
-      put("^\\d{1,2}/\\d{1,2}/\\d{4}$", "MM/dd/yyyy");
       put("^\\d{4}/\\d{1,2}/\\d{1,2}$", "yyyy/MM/dd");
+      put("^\\d{4}-\\d{1,2}-\\d{1,2}$", "yyyy-MM-dd");
       put("^\\d{1,2}\\s[a-z]{3}\\s\\d{4}$", "dd MMM yyyy");
       put("^\\d{1,2}\\s[a-z]{4,}\\s\\d{4}$", "dd MMMM yyyy");
       put("^\\d{1,2}-\\d{1,2}-\\d{2}$", "dd-MM-yy");
-      put("^\\d{1,2}/\\d{1,2}/\\d{2}$", "MM/dd/yy");
+      //try to catch "London's" issue
+      put("^(0[1-9]|[1-9][1-2]*)/\\d{1,2}/\\d{2}$", "MM/dd/yy");
+      put("^\\d{1,2}/(0[1-9]|[1-9][1-2]*)/\\d{2}$", "dd/MM/yy");
+      put("^(0[1-9]|[1-9][1-2]*)/\\d{1,2}/\\d{4}$", "MM/dd/yyyy");
+      put("^\\d{1,2}/(0[1-9]|[1-9][1-2]*)/\\d{4}$", "dd/MM/yyyy");
```

Comment:
I wonder if regexps are required at all.
Should ksar just try to parse the date and check which mask works better?